rust/protocol/src/session_cipher.rs

```diff
@@ -342,6 +401,20 @@ fn decrypt_message_with_record<R: Rng + CryptoRng>(
         record.promote_old_session(idx, updated_session)?;
         Ok(ptext)
     } else {
+        if let Ok(current_state) = record.session_state() {
+            log::error!(
+                "No valid session for recipient: {}, current session base key {}, number of previous states: {}",
+                remote_address,
+                hex::encode(current_state.sender_ratchet_key().unwrap().public_key_bytes().unwrap()),
+                record.previous_session_states().unwrap().count(),
+            );
+        } else {
+            log::error!(
+                "No valid session for recipient: {}, (no current session state), number of previous states: {}",
+                remote_address,
+                record.previous_session_states().unwrap().count(),
```

Comment:
Worth adding to `SessionState` a

`pub(crate) fn previous_session_state_count() -> usize`?

Reply:
Honestly either `previous_session_states` shouldn't be failable or it should be handled here.